cloudinit/config/schema.py

```diff
@@ -872,7 +916,20 @@ def validate_cloudconfig_file(
     if not isinstance(cloudconfig, dict):
         # Return a meaningful message on empty cloud-config
         if not annotate:
-            raise RuntimeError("Cloud-config is not a YAML dict.")
+            raise RuntimeError(
+                "Cloud-config[%s] %s is not a YAML dict."
+                % (schema_type, config_path)
+            )
+    if schema_type == "network-config":
+        # Pop optional top-level "network" key when present
+        netcfg = cloudconfig.get("network", cloudconfig)
+        if netcfg.get("version", 0) != 1:
```

Comment:
it will be there if an empty network dict is provided, or one without a mandatory 'version' key
```
$ cat net.yaml
network:
$ $ PYTHONPATH=. python3 -m cloudinit.cmd.main schema -c net.yaml --schema-type network-config
WARNING: datasource not detected, using default instance-data/user-data paths.
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/csmith/src/cloud-init/cloudinit/cmd/main.py", line 1108, in <module>
    return_value = main(sys.argv)
                   ^^^^^^^^^^^^^^
  File "/home/csmith/src/cloud-init/cloudinit/cmd/main.py", line 1094, in main
    retval = util.log_time(
             ^^^^^^^^^^^^^^
  File "/home/csmith/src/cloud-init/cloudinit/util.py", line 2834, in log_time
    ret = func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^
  File "/home/csmith/src/cloud-init/cloudinit/config/schema.py", line 1514, in handle_schema_args
    performed_schema_validation = validate_cloudconfig_file(
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/csmith/src/cloud-init/cloudinit/config/schema.py", line 924, in validate_cloudconfig_file
    if netcfg.get("version", 0) != 1:
       ^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get'
```

Reply:
Doesn't that require `if netcfg and netcfg.get("version") != 1:` then? If `netcfg` is `None`, having a default value for `.get()` won't help us.

Reply:
Ahh misread. Right, dropping default 0/None in this test and in the subsequent printed message. The default does helps nothing. I've also added a supplemental check and skip message for empty network config.